### PR TITLE
[keymgr] keymgr hardening part 2

### PR DESCRIPTION
**Pull request by tjaychen**
This PR adds a select few keymgr registers as "shadowed" variety.

### CODE REVIEW

**Review by eunchan — APPROVED** [review]
Thanks. It is the good example of how to implement shadow registers and connect alerts indeed. I will follow this example in PR #7439

**Comment by tjaychen** [comment]
> Thanks. It is the good example of how to implement shadow registers and connect alerts indeed. I will follow this example in PR #7439

gah! CI failed...maybe don't follow this example... hehe

File: hw/ip/keymgr/dv/env/keymgr_scoreboard.sv [hunk 1]
```diff
@@ -543,8 +544,10 @@ class keymgr_scoreboard extends cip_base_scoreboard #(
           end
         end
       end
-      "reseed_interval": begin
-        if (addr_phase_write) cfg.keymgr_vif.edn_interval = item.a_data;
+      "reseed_interval_shadowed": begin
+        //if (addr_phase_write) cfg.keymgr_vif.edn_interval = item.a_data;
+        if (addr_phase_write) cfg.keymgr_vif.edn_interval =
+          `gmv(ral.reseed_interval_shadowed.val);
```
Comment (imphil) [inline hw/ip/keymgr/dv/env/keymgr_scoreboard.sv:550]: Can you remove the commented-out line, and then format this as

```
if (addr_phase_write) 
  cfg.keymgr_vif.edn_interval = `gmv(ral.reseed_interval_shadowed.val);
```

File: hw/ip/keymgr/rtl/keymgr.sv [hunk 2]
```diff
@@ -372,7 +385,7 @@ module keymgr
   keymgr_key_dest_e cipher_sel;
   logic [KeyWidth-1:0] cipher_seed;
 
-  assign cipher_sel = keymgr_key_dest_e'(reg2hw.control.dest_sel);
+  assign cipher_sel = keymgr_key_dest_e'(reg2hw.control.dest_sel.q);
```
Comment (imphil) [inline hw/ip/keymgr/rtl/keymgr.sv:388]: Is this change related to the shadow registers?

**Review by imphil — APPROVED** [review]

File: hw/ip/keymgr/rtl/keymgr.sv (hunk 2 above)
Comment (tjaychen) [reply hw/ip/keymgr/rtl/keymgr.sv:388]: good catch. it "was".  I originally had `control` as a shadow register, then i realized our support for shadow register hrw type is .... weird.  So i reverted.

Let me remove the extra `.q`.

**Review by weicaiyang — APPROVED** [review]
LGTM
